open_instruct/grpo_fast.py

```diff
@@ -3072,17 +2929,8 @@ def health_check_fn():
 
     # Send initial data to ensure we have a N-step offset.
     for _ in range(args.async_steps * args.num_unique_prompts_rollout):
-        dataset_index = next(iter_dataloader)
-        add_prompt_to_generator(
-            train_dataset[dataset_index],
-            dataset_index,
-            iter_dataloader.epoch_number,
-            resume_training_step,
-            pending_queries_map,
-            param_prompt_Q,
-            generation_configs["train"],
-            is_eval=False,
-        )
+        example = next(data_loader_iter)
```

Comment:
usual question here: how do we want/did we decide to handle the case where the async steps go over an epoch boundary?